### ISSUE TITLE
Popover (and Dropdown) needs a way to set initial focus depending on its usage or content

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#4620]
Let's consider, for example, the "Content Structure" component (formerly "Table of Contents"). When clicking the toggle button that opens it, focus is moved to the first heading in the Document Outline. However, posts may have no headings at all:

<img width="594" alt="screen shot 2018-01-21 at 17 24 59" src="https://user-images.githubusercontent.com/1682452/35196733-5e896490-fed6-11e7-8e2c-e82af72370fd.png">

In this case, there's no element to move focus to and a focus loss occurs. Press Tab again and focus fallbacks to the document root:

<img width="610" alt="screen shot 2018-01-21 at 17 25 09" src="https://user-images.githubusercontent.com/1682452/35196752-7f9125d8-fed6-11e7-93e0-124e49aacea3.png">

Additionally, even if there were headings, all the first part of the content (the "counts") is skipped and users of assistive technologies would have to explore content backwards to get the counts.

In this specific case, the initial focus should probably be moved to the Popover container, and  the container should be properly labeled.

However, initial focus really depends on the content of the component. There might be cases where it would make sense to set the initial focus on a different element, for example an input field in a form with a very specific task to accomplish.

Since we can't predict which kind of content this component will be used for, the initial focus should be different depending on the specific component usage. I'd rather consider a way to set the initial focus via a prop, to pass in some way the element that should receive focus.

**Comment by afercia** [comment]
Totally agree thanks @grahamarmfield (hi!). About constraining tabbing, there's #5242

**Comment by noisysocks** [comment]
We should now be able to do this by setting `focusOnMount="container"` on the Popover. 

https://github.com/WordPress/gutenberg/blob/master/packages/components/src/popover/README.md#focusonmount
